doc/source/placement-group.rst

```diff
@@ -0,0 +1,201 @@
+
+.. _pgroup-strategy:
+
+Strategy types
+--------------
+
+Ray currently supports the following placement group strategies:
+
+**STRICT_PACK**
+
+All bundles must be placed into a single node on the cluster.
+
+**PACK**
+
+All provided bundles are packed onto a single node on a best-effort basis.
+If strict packing is not feasible (i.e., some bundles do not fit on the node), bundles can be placed onto other nodes nodes.
+
+**STRICT_SPREAD**
+
+Each bundle must be scheduled in a separate node.
+
+**SPREAD**
+
+Each bundle will be spread onto separate nodes on a best effort basis.
+If strict spreading is not feasible, bundles can be placed overlapping nodes.
+
+Quick Start
+-----------
+
+Let's see an example of using placement group. Note that this example is done within a single node.
+
+.. code-block:: python
+
+  import ray
+  from pprint import pprint
+
+  # Import placement group APIs.
+  from ray.util.placement_group import (
+      placement_group,
+      placement_group_table,
+      remove_placement_group
+  )
+
+  ray.init(num_gpus=2, resources={"extra_resource": 2})
+
+Let's create a placement group. Recall that each bundle is a collection of resources, and tasks or actors can be scheduled on each bundle.
+
+.. code-block:: python
+
+  gpu_bundle = {"GPU": 2}
+  extra_resource_bundle = {"extra_resource": 2}
+
+  # Reserve bundles with strict pack strategy.
+  # It means Ray will reserve 2 "GPU" and 2 "extra_resource" on the same node (strict pack) within a Ray cluster.
+  # Using this placement group for scheduling actors or tasks will guarantee that they will
+  # collocate in the same node.
+  pg = placement_group([gpu_bundle, extra_resource_bundle], strategy="STRICT_PACK")
+
+  # Wait until placement group is created.
+  ray.get(pg.ready())
+
+Now let's define an actor that uses GPU. We'll also define a task that use ``extra_resources``.
+
+.. code-block:: python
+
+  @ray.remote(num_gpus=1)
+  class GPUActor:
+      def __init__(self):
+          pass
+
+  @ray.remote(resources={"extra_resource": 1})
+  def extra_resource_task():
+      import time
+      # simulate long-running task.
+      time.sleep(10)
+
+  # Create GPU actors on a gpu bundle.
+  gpu_actors = [GPUActor.options(
+          placement_group=pg,
+          # This is the index from the original list.
+          placement_group_bundle_index=0) # Index of gpu_bundle is 0.
+      .remote() for _ in range(2)]
+
+  # Create extra_resource actors on a extra_resource bundle.
+  extra_resource_actors = [extra_resource_task.options(
+          placement_group=pg,
+          # This is the index from the original list.
+          placement_group_bundle_index=1) # Index of extra_resource_bundle is 1.
+      .remote() for _ in range(2)]
+
+Now, you can guarantee all gpu actor and extra_resource task are located on the same node
+because they are scheduled on a placement group with the STRICT_PACK strategy.
```

Comment:
```suggestion
Now, you can guarantee all gpu actors and extra_resource tasks are located on the same node
because they are scheduled on a placement group with the STRICT_PACK strategy.
```